src/App.tsx

```diff
@@ -1,26 +1,145 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoItem } from './components/TodoItem';
+import { FilterType } from './types/FilterType';
+import { ErrorTypes } from './types/ErrorTypes';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorTypes | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  const visibleTodos = useMemo(
+    () =>
+      todos.filter(todo => {
+        if (filterType === FilterType.All) {
+          return true;
+        }
+
+        return filterType === FilterType.Completed
+          ? todo.completed
+          : !todo.completed;
+      }),
+    [todos, filterType],
+  );
+
+  const inputField = useRef<HTMLInputElement>(null);
+
+  const isAllCompleted = todos.every(todo => todo.completed);
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const onAddTodo = async (newTodoTitle: string) => {
+    setIsSubmitting(true);
+    setTempTodo({
+      id: 0,
+      title: newTodoTitle,
+      completed: false,
+      userId: todoService.USER_ID,
+    });
+    try {
+      const newTodo = await todoService.addTodo({
+        title: newTodoTitle,
+        completed: false,
+        userId: todoService.USER_ID,
+      });
+
+      setTodos(currentTodos => [...currentTodos, newTodo]);
+    } catch (error) {
+      setErrorMessage(ErrorTypes.Adding);
+      inputField?.current?.focus();
+      throw error;
+    } finally {
+      setIsSubmitting(false);
+      setTempTodo(null);
+    }
+  };
+
+  const onDeleteTodo = async (todoToDelete: number) => {
+    setLoadingTodoIds(prev => [...prev, todoToDelete]);
+    try {
+      await todoService.deleteTodo(todoToDelete);
+      setTodos(prev => prev.filter(todo => todo.id !== todoToDelete));
+    } catch (error) {
+      setErrorMessage(ErrorTypes.Deleting);
+      inputField?.current?.focus();
+      throw error;
```

Comment:
You should not throw an error here, because you don't use `try/catch` in places where you use this function